### PR TITLE
Add unique constraint to database plugin names

### PR DESCRIPTION
**Pull request by Donnype**
### Changes

Add unique constraint to database plugin names and align API behavior

### Issue link

Closes https://github.com/minvws/nl-kat-coordination/issues/3547

### Demo

-
### QA notes

Please check you cannot create two plugins with the same name from Rocky!

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [x] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: boefjes/boefjes/sql/session.py [hunk 1]
```diff
@@ -34,15 +34,13 @@ def __exit__(self, exc_type: type[Exception], exc_value: str, exc_traceback: str
...
             return
 
-        error = None
-
         try:
             logger.debug("Committing session")
             self.session.commit()
-        except DatabaseError as e:
-            error = e
+        except exc.IntegrityError as e:
```
Comment (underdarknl) [inline boefjes/boefjes/sql/session.py:40]: Is there no more specific error for the uniqueConstraint being thrown? I see psycog2 has one specifically for that.

File: boefjes/tests/integration/test_api.py
Comment (ammar92) [inline boefjes/tests/integration/test_api.py:1]: Can you please use `model_dump_json()` instead of `json()`, so we don't have to do it in #3557

**Review by ammar92 — COMMENTED** [review]
Tiny remark and good point from @underdarknl. Looks good otherwise 👍

**Comment by Donnype** [comment]
@stephanie0x00 Thanks! And: handling of this error in Rocky will be handled in several places in #3566

File: boefjes/boefjes/sql/session.py (hunk 1 above)
Comment (Donnype) [reply boefjes/boefjes/sql/session.py:41]: I fixed this partly and double-checked the documentation, but the best we could do getting the field name out of the message is parsing the exception string I'm afraid

**Review by underdarknl — APPROVED** [review]

**Pull request merged**